core/src/main/java/org/apache/iceberg/rest/RESTClient.java

```diff
@@ -28,13 +29,15 @@
  */
 public interface RESTClient extends Closeable {
```

Comment:
If we do decide to define a `RequestContext`, this might be a good place to add an inner interface.